lib/net.ts

```diff
@@ -197,21 +198,54 @@ export class NetRequest {
 	}
 
 	/**
-	 * Makes a basic http/https request to the URI.
-	 * Returns the response data.
+	 * Makes a basic http/https request to the URI and returns the response data.
+	 * Automatically follows up to `opts.maxRedirects` (default 5) for HTTP 30x
+	 * redirects.
 	 *
-	 * Will throw if the response code isn't 200 OK.
+	 * Will throw if the final response code isn't 200 OK.
 	 *
-	 * @param opts request opts - headers, etc.
+	 * @param opts Request options - headers, body, timeout, etc.
 	 */
-	async get(opts: NetRequestOptions = {}): Promise<string> {
-		const stream = this.getStream(opts);
-		const response = await stream.response;
-		if (response) this.response = response;
-		if (response && response.statusCode !== 200) {
-			throw new HttpError(response.statusMessage || "Connection error", response.statusCode, await stream.readAll());
+	async get(opts: NetRequestOptions & { maxRedirects?: number } = {}): Promise<string> {
+		const maxRedirects = opts.maxRedirects ?? 5;
+		let currentURL: string = this.uri;
+		let redirectsRemaining = maxRedirects;
+
+		const currentOpts: NetRequestOptions & { maxRedirects?: number } = { ...opts };
+
+		while (true) {
+			const stream = new NetStream(currentURL, currentOpts);
+			const response = await stream.response;
+			if (response) this.response = response;
+
+			if (response) {
```

Comment:
This code will infinitely loop if there's no response, from the looks of this. De-incrementing the loop counter only in an if statement is dangerous. This entire function needs a rewrite.